### PR TITLE
Fix paligemma inverted mask

### LINKED ISSUE #31171: transformers 4.41.2 breaks paligemma inference
### System Info

Custom 4D attention mask should be passed in inverted form with max==0`

```
inputs = processor(prompt, raw_image, return_tensors="pt").to(device)
output = model.generate(**inputs, max_new_tokens=100)
```

breaks on the newest version with error: `Custom 4D attention mask should be passed in inverted form with max==0``

If i use transformers==4.41.1 things work as expected

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

inputs = processor(prompt, raw_image, return_tensors="pt").to(device)
output = model.generate(**inputs, max_new_tokens=100)

### Expected behavior

model should generate an output

### PR DESCRIPTION
**Pull request by molbap**
# What does this PR do?

This PR makes sure the causal mask created in paligemma merging is inverted. Else, inference will be broken by a `ValueError` check raised by `GemmaModel`.

In addition, a new test is added to make sure base fine-tuning setup is robust. 

Fixes #31171 

## Before submitting

- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Comment by molbap** [comment]
cc @ArthurZucker for review :)

**Review by satishlokkoju — APPROVED** [review]

**Review by ArthurZucker — APPROVED** [review]
Thanks for adding a test. 
cc @amyeroberts I think the fetcher did not trigger for PalIGemma when we modified Gemma, which is expected in a way since we use AutoModel.from_config
